stdlib/2and3/_types.pyi

```diff
@@ -0,0 +1,7 @@
+from typing import Protocol, Union
+
+class _HasFileno(Protocol):
+    def fileno(self) -> int: ...
+
+_FileDescriptor = int
+_FileObject = Union[int, _HasFileno]
```

Comment:
This is an unfortunate name because to me a "file object" is the return type of open(), not an int. Maybe `FileDescriptorLike` by analogy with `PathLike`?